src/langchain_google_spanner/loader.py

```diff
@@ -0,0 +1,422 @@
+            raise Exception("Use one of 'text', 'JSON', 'YAML', 'CSV'.")
+        self.databoost = databoost
+        self.client = client_with_user_agent(client, USER_AGENT_LOADER)
+        self.staleness = staleness
+        instance = self.client.instance(instance_id)
+        if not instance.exists():
+            raise Exception("Instance doesn't exist.")
+        database = instance.database(database_id)
+        if not database.exists():
+            raise Exception("Database doesn't exist.")
+
+    def load(self) -> List[Document]:
+        """
+        Load langchain documents from a Spanner database.
+
+        Returns:
+            (List[langchain_core.documents.Document]): a list of Documents with metadata
+            from specific columns.
+        """
+        return list(self.lazy_load())
+
+    def lazy_load(self) -> Iterator[Document]:
+        """
+        A lazy loader for langchain documents from a Spanner database. Use lazy load to avoid
+        caching all documents in memory at once.
+
+        Returns:
+            (Iterator[langchain_core.documents.Document]): a list of Documents with metadata
+            from specific columns.
+        """
+        instance = self.client.instance(self.instance_id)
+        db = instance.database(self.database_id)
+        timestamp = (
+            self.staleness.replace(tzinfo=datetime.timezone.utc)
+            if type(self.staleness) is datetime.datetime
+            else None
+        )
+        duration = (
+            datetime.timedelta(seconds=self.staleness)
+            if type(self.staleness) is float
+            else None
+        )
+        snapshot = db.batch_snapshot(exact_staleness=duration, read_timestamp=timestamp)
+        partitions = snapshot.generate_query_batches(
+            sql=self.query, data_boost_enabled=self.databoost
+        )
+        for partition in partitions:
+            r = snapshot.process_query_batch(partition)
+            results = r.to_dict_list()
+            if len(results) == 0:
+                break
+            column_names = [f.name for f in r.fields]
+            content_columns = self.content_columns or [column_names[0]]
+            metadata_columns = self.metadata_columns or [
+                col for col in column_names if col not in content_columns
+            ]
+            for row in results:
+                yield _load_row_to_doc(
+                    self.format,
+                    content_columns,
+                    metadata_columns,
+                    self.metadata_json_column,
+                    row,
+                )
+
+
+class SpannerDocumentSaver:
+    """Save docs to Google Cloud Spanner."""
+
+    def __init__(
+        self,
+        instance_id: str,
+        database_id: str,
+        table_name: str,
+        content_column: str = CONTENT_COL_NAME,
+        metadata_columns: List[str] = [],
+        metadata_json_column: str = METADATA_COL_NAME,
+        client: Optional[Client] = None,
```

Comment:
not assigning any default value?

Reply:
It will be assigned in the function `client_with_user_agent`.